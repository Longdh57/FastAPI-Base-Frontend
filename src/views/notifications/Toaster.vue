<template>
  <CCard>
    <CCardHeader>
      <div>
        <CIcon name="cil-task"/> Toasts
        <a 
          href="https://coreui.io/pro/vue/" 
          rel="noreferrer noopener" 
          target="_blank" 
          class="badge badge-danger ml-1"
        >
          CoreUI Pro
        </a>
        <div class="card-header-actions">
          <a 
            href="https://coreui.io/vue/docs/components/toast" 
            class="card-header-action" 
            rel="noreferrer noopener" 
            target="_blank"
          >
            <small class="text-muted">docs</small>
          </a>
        </div>
      </div>
    </CCardHeader>
    <CRow>
      <CCol sm="12" lg="6">
        <CForm class="m-4">
          <h5>Add toast with following props:</h5>
          <CInputCheckbox
            :checked.sync="autohide"
            custom
            label="Autohide of the toast"
            class="my-2 mt-4"
          />
          <CInput
            v-if="autohide"
            :value.sync="autohideValue"
            type="number"
            lazy
            label="Time to autohide"
            class="my-2"
          />
          <CSelect
            :value.sync="position"
            :options="positions"
            custom
            label="Position"
            class="my-2"
          />

          <CInputCheckbox
            :checked.sync="fade"
            custom
            label="fade"
            class="my-2"
          />

          <CInputCheckbox
            :checked.sync="closeButton"
            custom
            label="closeButton"
            class="my-2"
          />

          <CButton 
            class="mr-1 w-25" 
            color="success" 
            @click="addToast()"
          >
            Add toast
          </CButton>
        </CForm>
      </CCol>
      <CCol sm="12" lg="6">
        <div class="m-4">
          <CToaster 
            v-for="(toaster, toasterKey) in toasters"
            :position="toasterKey"
            :key="'toaster' + toasterKey"
          >
            <template v-for="(toast, key) in toaster">
              <CToast
                :key="'toast' + key"
                :show="true"
                v-bind="toast"
              >
                {{`This is a toast in ${toasterKey} positioned toaster number ${key + 1}.`}}
              </CToast>
            </template>
          </CToaster>
        </div>
      </CCol>
    </CRow>
  </CCard>
</template>

<script>

export default {
  name: 'Toaster',
  data () {
    return {
      toasts: [
        { position: 'static', header: 'Toast title'},
        { position: 'static', header: 'Toast title'},
        { position: 'top-right', header: 'Toast title', autohide: 3000 }
      ],
      autohide: true,
      autohideValue: 5000,
      closeButton: true,
      fade: true,
      position: 'top-right',
      header: 'Toast title',
      positions: [
        'static',
        'top-left', 
        'top-center', 
        'top-right', 
        'top-full',
        'bottom-left',
        'bottom-center', 
        'bottom-right',
        'bottom-full'
      ]
    }
  },
  computed: {
    toastConfig () {
      return {
        autohide: this.autohide ? this.autohideValue : false,
        closeButton: this.closeButton,
        fade: this.fade,
        position: this.position,
        header: this.header
      }
    },
    toasters () {
      return this.toasts.reduce((toasters, toast) => {
        toasters[toast.position] = toasters[toast.position] || []
        toasters[toast.position].push(toast)
        return toasters
      }, {})
    }
  },
  methods: {
    addToast () {
      this.toasts.push(this.toastConfig)
    }
  }
}
</script>
