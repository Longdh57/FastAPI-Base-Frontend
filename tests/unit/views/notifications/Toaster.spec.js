import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue-pro'
import Toaster from '@/views/notifications/Toaster'

Vue.use(CoreuiVue)

describe('Toaster.vue', () => {
  it('has a name', () => {
    expect(Toaster.name).toMatch('Toaster')
  })
  it('has a created hook', () => {
    expect(typeof Toaster.data).toMatch('function')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Toaster)
    expect(wrapper.vm).toBeTruthy()
  })
  it('is Toastr', () => {
    const wrapper = shallowMount(Toaster)
    expect(wrapper.findComponent(Toaster)).toBeTruthy()
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Toaster)
    expect(wrapper.element).toMatchSnapshot()
  })
})
