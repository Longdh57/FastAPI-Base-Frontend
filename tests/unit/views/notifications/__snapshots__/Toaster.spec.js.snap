// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster.vue renders correctly 1`] = `
<ccard-stub>
  <ccardheader-stub>
    <div>
      <cicon-stub
        name="cil-task"
      />
       Toasts
      
      <a
        class="badge badge-danger ml-1"
        href="https://coreui.io/pro/vue/"
        rel="noreferrer noopener"
        target="_blank"
      >
        
        CoreUI Pro
      
      </a>
       
      <div
        class="card-header-actions"
      >
        <a
          class="card-header-action"
          href="https://coreui.io/vue/docs/components/toast"
          rel="noreferrer noopener"
          target="_blank"
        >
          <small
            class="text-muted"
          >
            docs
          </small>
        </a>
      </div>
    </div>
  </ccardheader-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      lg="6"
      sm="12"
      tag="div"
    >
      <cform-stub
        class="m-4"
      >
        <h5>
          Add toast with following props:
        </h5>
         
        <cinputcheckbox-stub
          checked="true"
          class="my-2 mt-4"
          custom="true"
          label="Autohide of the toast"
        />
         
        <cinput-stub
          class="my-2"
          label="Time to autohide"
          lazy="true"
          type="number"
          value="5000"
        />
         
        <cselect-stub
          class="my-2"
          custom="true"
          label="Position"
          options="static,top-left,top-center,top-right,top-full,bottom-left,bottom-center,bottom-right,bottom-full"
          value="top-right"
        />
         
        <cinputcheckbox-stub
          checked="true"
          class="my-2"
          custom="true"
          label="fade"
        />
         
        <cinputcheckbox-stub
          checked="true"
          class="my-2"
          custom="true"
          label="closeButton"
        />
         
        <cbutton-stub
          activeclass="router-link-active"
          class="mr-1 w-25"
          color="success"
          event="click"
          exactactiveclass="router-link-exact-active"
          routertag="a"
          target="_self"
          type="button"
        >
          
          Add toast
        
        </cbutton-stub>
      </cform-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="6"
      sm="12"
      tag="div"
    >
      <div
        class="m-4"
      >
        <ctoaster-stub
          closebutton="true"
          fade="true"
          position="static"
        >
          <ctoast-stub
            closebutton="true"
            fade="true"
            header="Toast title"
            position="static"
            show="true"
          >
            
              This is a toast in static positioned toaster number 1.
            
          </ctoast-stub>
          <ctoast-stub
            closebutton="true"
            fade="true"
            header="Toast title"
            position="static"
            show="true"
          >
            
              This is a toast in static positioned toaster number 2.
            
          </ctoast-stub>
        </ctoaster-stub>
        <ctoaster-stub
          closebutton="true"
          fade="true"
          position="top-right"
        >
          <ctoast-stub
            autohide="3000"
            closebutton="true"
            fade="true"
            header="Toast title"
            position="top-right"
            show="true"
          >
            
              This is a toast in top-right positioned toaster number 1.
            
          </ctoast-stub>
        </ctoaster-stub>
      </div>
    </ccol-stub>
  </crow-stub>
</ccard-stub>
`;
